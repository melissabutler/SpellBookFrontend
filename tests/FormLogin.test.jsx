import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import LoginForm from "../src/Components/FormLogin"

describe("LoginForm", function() {
        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({ currentUser: null}),
                useRef: () => { target: null }
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            const navigate = vi.fn()
            return {
                ...mod, 
                useNavigate: () => navigate,
                BrowserRouter: () => "",
                Link: () => "",

            }
        })
        it('renders the LoginForm component', async () => {
            render(<LoginForm />)
            screen.debug();
            expect(screen.getByText('Username')).toBeInTheDocument();
        
        });

        it('matches snapshot', () => {
            const { asFragment } = render(<LoginForm />);
            expect(asFragment()).toMatchSnapshot();
        });



})
   