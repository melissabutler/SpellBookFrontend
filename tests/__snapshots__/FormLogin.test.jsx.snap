// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginForm > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form container"
  >
    <div
      class="row"
    >
      <h1>
        Log In
      </h1>
    </div>
    <form
      class="Form-form"
    >
      <div
        class="Form-section-label row"
      >
        <label
          class="Form-label"
          for="username"
        >
          Username
        </label>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <input
            class="Form-input"
            id="username"
            name="username"
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="Form-section-label row"
      >
        <label
          class="Form-label"
          for="password"
        >
          Password
        </label>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <input
            class="Form-input"
            id="password"
            name="password"
            placeholder="Password"
            type="text"
            value=""
          />
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <button>
            Log In
          </button>
        </div>
        <div
          class="col"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
